shared/events/amplitude/publisher.py

```
@@ -48,6 +53,31 @@ def __init__(self, override_env=False):
 
     def publish(
         self, event_type: AmplitudeEventType, event_properties: AmplitudeEventProperties
+    ):
+        inc_counter(
+            AMPLITUDE_PUBLISH_COUNTER,
+            labels={
+                "event_type": event_type,
+            },
+        )
+        try:
+            self._unsafe_publish(event_type, event_properties)
+        except Exception as e:
+            inc_counter(
+                AMPLITUDE_PUBLISH_FAILURE_COUNTER,
+                labels={"event_type": event_type, "error": e.__class__.__name__},
+            )
+            log.error(
+                "Failed to publish Amplitude event",
+                extra=dict(
+                    event_type=event_type,
+                    error_name=e.__class__.__name__,
+                    error_message=str(e),
+                ),
+            )
+
```

`except Exception` is broad - every type of raised error will qualify. Can you make it more specific?
If not, it looks like you're swallowing the error (nothing is raised) - is this how you want it to work?

---

This is the intention yes. The only code below this is the little logic in this file and the underlying amplitude library. We are adding the error details to the log and the counter, so in theory we can catch unexpected errors getting swallowed here. 

The end goal of this PR is to make certain calling `publish` will never error out so we can put it wherever we want with 0 risk of interrupting critical flows.

---

I see, that makes sense. awesome!